daemon/graphdriver/vfs/driver.go

```diff
@@ -30,13 +33,22 @@ func Init(home string, options []string, uidMaps, gidMaps []idtools.IDMap) (grap
 		home:      home,
 		idMapping: idtools.NewIDMappingsFromMaps(uidMaps, gidMaps),
 	}
+
+	if err := d.parseOptions(options); err != nil {
+		return nil, err
+	}
+
 	rootIDs := d.idMapping.RootPair()
 	if err := idtools.MkdirAllAndChown(home, 0700, rootIDs); err != nil {
 		return nil, err
 	}
 
 	setupDriverQuota(d)
 
+	if size := d.getQuotaOpt(); !d.quotaSupported() && size > 0 {
+		return nil, errdefs.InvalidParameter(errors.Errorf("storage opt size no supported for vfs: %v", quota.ErrQuotaNotSupported))
```

Comment:
small typo here; 

```suggestion
		return nil, errdefs.InvalidParameter(errors.Errorf("storage opt size not supported for vfs: %v", quota.ErrQuotaNotSupported))
```


Looks like `quota.ErrQuotaNotSupported` already implements [`errdef.ErrNotImplemented`](https://github.com/moby/moby/blob/8e610b2b55bfd1bfa9436ab110d311f5e8a74dcb/errdefs/defs.go#L51-L54). https://github.com/moby/moby/blob/8e610b2b55bfd1bfa9436ab110d311f5e8a74dcb/daemon/graphdriver/quota/errors.go#L5-L19


So, this changes an "Not supported" error into an "Invalid parameter". I _guess_ that makes sense (as  it's an invalid configuration), but wondering now if `ErrQuotaNotSupported` should be an "invalid parameter" in other cases as well.

However, it we _do_ want to change this in an InvalidParameter here, I think this should probably not use `%v` to included the error, but "wrap" it instead; given that the original error already points out what the problem is ("Filesystem does not support, or has not enabled quotas") perhaps this could just be;

```suggestion
		return nil, errdefs.InvalidParameter(quota.ErrQuotaNotSupported)
```

or alternatively;

```suggestion
		return nil, errdefs.InvalidParameter(errors.Wrap(quota.ErrQuotaNotSupported, "storage opt size not supported for vfs"))
```

Reply:
Actually; isn't this already validated by `parseOptions` above?

Reply:
parseOptions only validate the validity of parameters, and cannot confirm whether Quota is supported or not.